molgenis-ontology/src/main/java/org/molgenis/ontology/controller/SortaServiceAnonymousController.java

```diff
@@ -0,0 +1,290 @@
+package org.molgenis.ontology.controller;
+
+import static org.molgenis.ontology.controller.SortaServiceAnonymousController.URI;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.text.DecimalFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.Part;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.Entity;
+import org.molgenis.data.Repository;
+import org.molgenis.data.csv.CsvRepository;
+import org.molgenis.data.csv.CsvWriter;
+import org.molgenis.data.processor.CellProcessor;
+import org.molgenis.data.processor.LowerCaseProcessor;
+import org.molgenis.data.processor.TrimProcessor;
+import org.molgenis.data.support.MapEntity;
+import org.molgenis.file.FileStore;
+import org.molgenis.ontology.core.meta.OntologyTermMetaData;
+import org.molgenis.ontology.sorta.MatchingTaskEntityMetaData;
+import org.molgenis.ontology.sorta.SortaModifiableCsvRepository;
+import org.molgenis.ontology.sorta.SortaService;
+import org.molgenis.ontology.sorta.SortaServiceImpl;
+import org.molgenis.ontology.utils.SortaServiceUtil;
+import org.molgenis.ui.MolgenisPluginController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
+@Controller
+@RequestMapping(URI)
+public class SortaServiceAnonymousController extends MolgenisPluginController
```

Comment:
I will look at the possibility of giving the anonymous user the right permissions to use SORTA without exposing other information. Then we can happily remove the SORTA_Anonymous controller